org.eclipse.lsp4e/src/org/eclipse/lsp4e/ConnectDocumentToLanguageServerSetupParticipant.java

```diff
@@ -34,7 +36,7 @@
  *
  */
 public class ConnectDocumentToLanguageServerSetupParticipant implements IDocumentSetupParticipant, IDocumentSetupParticipantExtension {
-	private static final WeakHashMap<CompletableFuture<?>, Void> PENDING_CONNECTIONS = new WeakHashMap<>();
+	private static final Cache<CompletableFuture<?>, Void> PENDING_CONNECTIONS = CacheBuilder.newBuilder().weakKeys().build();
```

Comment:
Would it not be more standard to wrap it as in `Collections.synchronizedMap(new WeakHashMap<>())`?

I have checked the CacheBuilder and I think in the end it will something very similar, but it is harder to understand, as one needs to know Guava, which is an extra API on top of standard Java.